features/flows/headers/vxlan/test_vxlan_header_with_inner_ipv6_traffic.py

```diff
@@ -0,0 +1,188 @@
+    eth2.dst.value = tc["innerRxMac"]
+
+    ip.src.value = tc["txIp"]
+    ip.dst.value = tc["rxIp"]
+
+    udp.src_port.value = tc["txUdpPortValue"]
+    udp.dst_port.value = tc["rxUdpPortValue"]
+
+    vxlan.vni.values = tc["vxLanVniValues"]
+
+    tcp.src_port.value = tc["txTcpPortValue"]
+    tcp.dst_port.value = tc["rxTcpPortValue"]
+
+    log.info("Config:\n%s", c)
+    return c
+
+
+def metrics_ok(api, tc):
+    m = api.get_flow_metrics()[0]
+    ok = (
+        m.transmit == m.STOPPED
+        and m.frames_tx == tc["pktCount"]
+        and m.frames_rx == tc["pktCount"]
+    )
+    return ok
+
+
+def capture_ok(api, c, tc):
+    if not api.test_config.otg_capture_check:
+        return
+
+    ignored_count = 0
+    captured_packets = api.get_capture(c.ports[1].name)
+
+    for i, p in enumerate(captured_packets.packets):
+
+        # ignore unexpected packets based on ethernet src MAC
+        if not captured_packets.has_field(
+            "ethernet src", i, 6, api.mac_addr_to_bytes(tc["txMac"])
+        ):
+            ignored_count += 1
+            continue
+
+        # packet size
+        captured_packets.validate_size(i, tc["pktSize"])
+
+        # ethernet header
+        captured_packets.validate_field(
+            "ethernet dst", i, 0, api.mac_addr_to_bytes(tc["rxMac"])
+        )
+        captured_packets.validate_field(
+            "ethernet type", i, 12, api.num_to_bytes(2048, 2)
+        )
+
+        # ipv4 header
+        captured_packets.validate_field(
+            "ipv4 total length", i, 16, api.num_to_bytes(tc["pktSize"] - 14 - 4, 2)
+        )
+        captured_packets.validate_field("ipv4 protocol", i, 23, api.num_to_bytes(17, 1))
+
+        # udp header
+        captured_packets.validate_field(
+            "udp src", i, 34, api.num_to_bytes(tc["txUdpPortValue"], 2)
+        )
+        captured_packets.validate_field(
+            "udp dst",
+            i,
+            36,
+            api.num_to_bytes(
+                tc["rxUdpPortValue"],
+                2,
+            ),
+        )
+        captured_packets.validate_field(
+            "udp length", i, 38, api.num_to_bytes(tc["pktSize"] - 14 - 4 - 20, 2)
+        )
+
+        # vxlan header
+        j = i - ignored_count
+        captured_packets.validate_field(
+            "vxlan Network Identifier",
+            i,
+            46,
+            api.num_to_bytes(tc["vxLanVniValues"][j % len(tc["vxLanVniValues"])], 3),
+        )
+
+        # inner ethernet header
+        captured_packets.validate_field(
+            "ethernet dst", i, 50, api.mac_addr_to_bytes(tc["innerRxMac"])
+        )
+        captured_packets.validate_field(
+            "ethernet type", i, 62, api.num_to_bytes(34525, 2)
+        )
+
+        # inner ipv6 header
+        captured_packets.validate_field(
+            "ipv6 payload length",
+            i,
+            68,
+            api.num_to_bytes(tc["pktSize"] - 14 - 4 - 20 - 8 - 8 - 14 - 40, 2),
```

Comment:
## Observation 
I saw that the payload length of this particular packet is coming out to be 148

Now I assume it should come out to be 144 

Calculation to support that the above statement
eth(14) + eth-payload(4) + ip(20) + udp(8) + vxlan(8) + inner-eth(14) + inner-eth-payload(4) + ipv6-header(40) = 112
Total packet(size) 126 - 112 = 144 (I see this value in Ixn Packet editor as well)

For now i was not sure for the calculation so pushed the changes with 148 (if theres is a bug I guess the inner eths payload is not being taken into the account maybe ?)

Reply:
I just noticed that in test you're using:
```
api.num_to_bytes(tc["pktSize"] - 14 - 4 - 20 - 8 - 8 - 14 - 40, 2)
```

you should instead use:
```
api.num_to_bytes(tc["pktSize"] - 14 - 20 - 8 - 8 - 14 - 40, 4)
```

Reply:
issue opened 
https://github.com/open-traffic-generator/ixia-c/issues/112